samples/seat_massager/consumer/src/main.rs

```diff
@@ -0,0 +1,193 @@
+use serde_derive::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use tokio::time::{sleep, Duration};
+use tonic::transport::Server;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Property {
+    #[serde(rename = "MassageAirbags")]
+    massage_airbags: sdv::airbag_seat_massager::massage_airbags::TYPE,
+    #[serde(rename = "$metadata")]
+    metadata: Metadata,
+}
+
+/// Start the seat massage sequence.
+///
+/// # Arguments
+/// `provider_uri` - The provider uri.
+fn start_seat_massage_sequence(provider_uri: String) {
+    debug!("Starting the consumer's seat massage sequence.");
+
+    let mut crest_row: i8 = 0;
+    let mut is_wave_moving_forwards = true;
+    const MAX_ROW: i8 = 5;
+
+    let metadata: Metadata =
+        Metadata { model: sdv::airbag_seat_massager::massage_airbags::ID.to_string() };
+    let mut property: Property = Property { massage_airbags: Vec::new(), metadata };
+
+    tokio::spawn(async move {
+        loop {
+            // We assume that the seat has 5 rows with 2 airbags in each row.
+            // The sequence will represent a wave with the crest_row representing the crest of the wave.
+            property.massage_airbags = Vec::new();
+            for airbag in 0..18 {
+                let row = airbag / 3;
+                let rows_from_crest = i8::abs(crest_row - row);
+                let mut inflation_level = 100 - (rows_from_crest * 20);
+                if inflation_level < 0 {
+                    inflation_level = 0;
+                }
+                property.massage_airbags.push(inflation_level as i32);
+            }
+
+            let value = serde_json::to_string_pretty(&property).unwrap();
+
+            info!(
+                "Sending a set request for entity id {} to provider URI {provider_uri}",
+                sdv::airbag_seat_massager::massage_airbags::ID
+            );
+
+            let client_result = DigitalTwinProviderClient::connect(provider_uri.clone()).await;
+            if client_result.is_err() {
+                warn!("Unable to connect. We will retry in a moment.");
+                sleep(Duration::from_secs(1)).await;
+                continue;
+            }
+            let mut client = client_result.unwrap();
+
+            let request = tonic::Request::new(SetRequest {
+                entity_id: sdv::airbag_seat_massager::massage_airbags::ID.to_string(),
+                value,
+            });
+
+            let response = client.set(request).await;
+            if let Err(status) = response {
+                warn!("{status:?}");
+            }
+
+            // Set crest_row for the next loop iteration.
+            if is_wave_moving_forwards {
+                if crest_row == MAX_ROW {
+                    crest_row = MAX_ROW - 1;
+                    is_wave_moving_forwards = false;
+                } else {
+                    crest_row += 1;
+                }
+            } else if crest_row == 0 {
```

Comment:
for readability and consistency between the cases I would write this loop like this:

```rust
if is_wave_moving_forwards {
    if crest_row == MAX_ROW {
        crest_row -= 1;
        is_wave_moving_forwards = false;
    } else {
        crest_row += 1;
    }
} else {
    if crest_row == 0 {
        crest_row += 1;
        is_wave_moving_forwards = true;
    } else {
        crest_row -= 1;
    }
}
```

Alternatively, this should have the same behavior and might be easier to read without the nesting:

```rust
if crest_row == 0 {
    is_wave_moving_forwards = true;
} else if crest_row = MAX_ROW {
    is_wave_moving_forwards = false;
}

if is_wave_moving_forwards {
    crest_row += 1;
} else {
    crest_row -= 1;
}

Reply:
Looks good.  I'll use it.